### PR TITLE
Pinning starsessions to 1.x

### PR DESCRIPTION
**Pull request by hhsecond**
## What does this PR do?

The recent release of [starsessions 2.0](https://github.com/alex-oleshkevich/starsessions) has broken lightning app as some of the arguments (evidently `secret_key`) are removed. 
Starsessions has made a major release which broke lightning app as we didn't have it pinned.

## Impact
- Anyone is trying to install the lightning from pypi starting yesterday to run a LightningApp won't be able to run an app even locally.
- Every lightning app that is created from yesterday in the cloud will also be failed

This PR pins the starsessions to 1.3 which is the known working version of starsessions with lightningapp. We'd need to unpin when we work on sessions again

This PR also fixes a bug in our setup tools that prevents our internal `# strict` parameter being considered.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @tchaton @rohitgr7

### CODE REVIEW

File: requirements/app/base.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 lightning-cloud==0.5.3
 packaging
 deepdiff>=5.7.0, <=5.8.1
-starsessions
+starsessions==1.3
```
Comment (Borda) [inline requirements/app/base.txt:4]: Can we give some range?

**Review by Borda — APPROVED** [review]

File: requirements/app/base.txt (hunk 1 above)
Comment (Borda) [reply requirements/app/base.txt:4]:
```suggestion
starsessions>=1.0, <2.0
```

Comment (hhsecond) [reply requirements/app/base.txt:4]: we should revert it soon and upgrade it to 2.0. I expect that to happen when we work on multitenancy.

File: requirements/app/cloud.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-starsessions
+starsessions>=1.0, <2.0 # strict
```
Comment (Borda) [inline requirements/app/cloud.txt:1]: This can be removed as it is already in base

**Review by Borda — APPROVED** [review]

**Review by lantiga — APPROVED** [review]

File: requirements/app/cloud.txt (hunk 2 above)
Comment (Borda) [reply requirements/app/cloud.txt:1]:
```suggestion
```

**Review by panos-is — APPROVED** [review]

**Comment by Borda** [comment]
@hhsecond mind chech also
```
E   ImportError: cannot import name 'Schema' from 'pydantic' 
```

**Review by dmitsf — APPROVED** [review]
Great, let's merge it to unblock CI

**Review by manskx — APPROVED** [review]
Thanks for fixing this !

Let's also explore how we can remove it to simplify our dependencies.

**Pull request merged**